app/src/main/java/com/example/photostomusic/LoginActivity.java

```diff
@@ -0,0 +1,136 @@
+package com.example.photostomusic;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.parse.LogInCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.spotify.sdk.android.authentication.AuthenticationClient;
+import com.spotify.sdk.android.authentication.AuthenticationRequest;
+import com.spotify.sdk.android.authentication.AuthenticationResponse;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+public class LoginActivity extends AppCompatActivity {
+
+    // Class name used as TAG for debugging
+    public final String TAG = this.getClass().getSimpleName();
+
+    // PSF variables required to interact with the Spotify Auth API
+    private static final String CLIENT_ID = "59a64c83ef024ea786df03a966505f91";
```

Comment:
does this need to be hardcoded here? could you not store the client ID in the DB?